src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -253,6 +253,17 @@ private static Field<Object> field(FieldCondition<?> condition, EntityType entit
       .orElseThrow(() -> new ColumnNotFoundException(entityType.getName(), columnName));
   }
 
+  // Suppress the unchecked cast warning on the Class<T> cast below. We need the correct type there in order to get
+  // a reasonable level of type-safety with the return type on this method
+  @SuppressWarnings("unchecked")
```

Comment:
I fought with this for a while and eventually just gave up on trying to do it right and just did a cast to `Class<T>` in the method.